### ISSUE TITLE
在Noita中按下任意按键均会导致游戏卡顿

### ISSUE THREAD

**Issue by nightson** [rime/weasel#1159]
**上报前请检查**
- [x] 我遇到的问题没有其他人在 issue 里提到过
- [x] 我的小狼毫版本于 rime/weasel 下载
- [x] 我在使用小狼毫的最新发布版本，或最新发布版本后的 CI 构建

**操作系统信息**
 - OS 详细版本: Windows 11 22H2 (OS Build 22621.3296)
 - 小狼毫版本: 0.15.0.0

**描述遇到的问题**
在Noita中按下任意按键均会导致游戏卡顿

**复现步骤**
重现问题的步骤，如：
1. 运行Noita游戏（about:blank）
2. 使用键盘进行任何操作
3. 游戏会异常卡顿

**预期行为**
游戏流畅运行（关闭算法服务后游戏一切正常）

**用户文件**

https://github.com/rime/weasel/assets/843241/e41f0351-f69c-4a49-a009-912845ca1001

**Comment by byzod** [comment]
哈……游戏正好没删，帮你试了试，win10 x64 21h2下无法复现哟，不管什么输入模式

另外我是noita beta
![An1](https://github.com/rime/weasel/assets/848607/31cac530-e31f-4524-9dc5-a82ffb7beef9)

**Comment by nightson** [comment]
> 哈……游戏正好没删，帮你试了试，win10 x64 21h2下无法复现哟，不管什么输入模式
> 
> 另外我是noita beta ![An1](https://private-user-images.githubusercontent.com/848607/318178811-31cac530-e31f-4524-9dc5-a82ffb7beef9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODkyNzAsIm5iZiI6MTcxMTc4ODk3MCwicGF0aCI6Ii84NDg2MDcvMzE4MTc4ODExLTMxY2FjNTMwLWUzMWYtNDUyNC05ZGM1LWE4MmZmYjdiZWVmOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODU2MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTM0MDNmM2I0ZmE1MjViZjc3NDI2ZGYxM2ZlMTFlMDZiNjgzZTVlM2MxYTFkMDBjNDdhNGU2YjMwYWY0MzI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OEp2-RRj1HnKS_S9usBFfPy5gZvo9oF1zDNXX-LeVCs) [ ![An1](https://private-user-images.githubusercontent.com/848607/318178811-31cac530-e31f-4524-9dc5-a82ffb7beef9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODkyNzAsIm5iZiI6MTcxMTc4ODk3MCwicGF0aCI6Ii84NDg2MDcvMzE4MTc4ODExLTMxY2FjNTMwLWUzMWYtNDUyNC05ZGM1LWE4MmZmYjdiZWVmOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODU2MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTM0MDNmM2I0ZmE1MjViZjc3NDI2ZGYxM2ZlMTFlMDZiNjgzZTVlM2MxYTFkMDBjNDdhNGU2YjMwYWY0MzI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OEp2-RRj1HnKS_S9usBFfPy5gZvo9oF1zDNXX-LeVCs) ](https://private-user-images.githubusercontent.com/848607/318178811-31cac530-e31f-4524-9dc5-a82ffb7beef9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODkyNzAsIm5iZiI6MTcxMTc4ODk3MCwicGF0aCI6Ii84NDg2MDcvMzE4MTc4ODExLTMxY2FjNTMwLWUzMWYtNDUyNC05ZGM1LWE4MmZmYjdiZWVmOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODU2MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTM0MDNmM2I0ZmE1MjViZjc3NDI2ZGYxM2ZlMTFlMDZiNjgzZTVlM2MxYTFkMDBjNDdhNGU2YjMwYWY0MzI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OEp2-RRj1HnKS_S9usBFfPy5gZvo9oF1zDNXX-LeVCs) [ ](https://private-user-images.githubusercontent.com/848607/318178811-31cac530-e31f-4524-9dc5-a82ffb7beef9.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3ODkyNzAsIm5iZiI6MTcxMTc4ODk3MCwicGF0aCI6Ii84NDg2MDcvMzE4MTc4ODExLTMxY2FjNTMwLWUzMWYtNDUyNC05ZGM1LWE4MmZmYjdiZWVmOS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQwODU2MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTM0MDNmM2I0ZmE1MjViZjc3NDI2ZGYxM2ZlMTFlMDZiNjgzZTVlM2MxYTFkMDBjNDdhNGU2YjMwYWY0MzI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OEp2-RRj1HnKS_S9usBFfPy5gZvo9oF1zDNXX-LeVCs)

感谢帮忙测试。看来是是只有在Windows 11下才可以复现的bug? 我在我的笔记本电脑上也可以复现，系统版本为Windows 11 Home 23H2。

**Comment by MoonSueAoi** [comment]
我在别的游戏（ff14）里貌似重现了这个问题，在中文模式下一直按一个键会让帧数持续下降，停按帧数就上来了，英文模式没这个问题

**Comment by adoyle-h** [comment]
不是 weasel 的问题，是 Noita 本身的问题。我使用搜狗输入法也是一样的，只要开了中文输入就会卡顿。解决方法就是确保没开启中文输入，使用英文输入法。
